Branch-SDK/Branch-SDK/BranchUniversalObject.m

```diff
@@ -0,0 +1,137 @@
+//
+//  BranchUniversalObject.m
+//  Branch-TestBed
+//
+//  Created by Derrick Staten on 10/16/15.
+//  Copyright © 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BranchUniversalObject.h"
+#import "BNCError.h"
+#import "BranchConstants.h"
+
+@implementation BranchUniversalObject
+
+- (instancetype)initWithCanonicalIdentifier:(NSString *)canonicalIdentifier {
+    if (self = [super init]) {
+        self.canonicalIdentifier = canonicalIdentifier;
+    }
+    return self;
+}
+
+- (instancetype)initWithTitle:(NSString *)title {
+    if (self = [super init]) {
+        self.title = title;
+    }
+    return self;
+}
+
+- (void)registerView {
+    Branch *branch = [Branch getCurrentInstanceIfAny];
+    if (!branch) {
+        NSLog(@"[Branch Warning] getInstance has not yet been invoked, so could not register view.");
+        return;
+    }
+    if (!self.canonicalIdentifier && !self.title) {
```

Comment:
same response as above. we always still need server-side validation, and the dev can get dirty data by passing in @"-" and @"-". so this is supposed to cover the general case but not be bulletproof on its own